### PR TITLE
Feat/amino input v3

### PR DESCRIPTION
**Pull request by Thienhuynh95**
## Jira issue

[Updates to form component designs](https://myzonos.atlassian.net/browse/BBT-464
## Description

This PR styles number, search & password input type

### CODE REVIEW

**Comment by goodpickle** [comment]
@Thienhuynh95 Looks like there is a merge conflict to resolve

File: src/components/Input/SearchInput.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      <StyledLabel htmlFor={AminoInput}>
+        <SearchIcon size={20} />
+      </StyledLabel>
+      <AminoInput
+        autoFocus={autoFocus}
+        disabled={disabled}
+        inputMode={inputMode}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        pattern={pattern && pattern}
```
Comment (goodpickle) [inline src/components/Input/SearchInput.tsx:91]:
```suggestion
        pattern={pattern}
```

File: src/components/Input/SearchInput.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        placeholder={placeholder || 'search...'}
+        readOnly={readOnly || false}
+        required={required}
+        tabIndex={tabIndex && tabIndex}
```
Comment (goodpickle) [inline src/components/Input/SearchInput.tsx:95]:
```suggestion
        tabIndex={tabIndex}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+          disabled={disabled}
+          inputMode={inputMode}
+          onChange={onChange}
+          onKeyDown={onKeyDown && onKeyDown}
+          pattern={pattern && pattern}
+          placeholder={placeholder || label}
+          readOnly={readOnly || false}
+          ref={ref}
+          required={required || false}
+          tabIndex={tabIndex && tabIndex}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:156]:
```suggestion
          tabIndex={tabIndex}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            error && error.length ? 'has-error' : '',
+            label ? 'has-label' : '',
+            value ? 'has-content' : '',
+          ].join(' ')}
+          autoFocus={autoFocus && autoFocus}
+          disabled={disabled}
+          inputMode={inputMode}
+          onChange={onChange}
+          onKeyDown={onKeyDown && onKeyDown}
+          pattern={pattern && pattern}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:151]:
```suggestion
          pattern={pattern}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+          className={[
+            error && error.length ? 'has-error' : '',
+            label ? 'has-label' : '',
+            value ? 'has-content' : '',
+          ].join(' ')}
+          autoFocus={autoFocus && autoFocus}
+          disabled={disabled}
+          inputMode={inputMode}
+          onChange={onChange}
+          onKeyDown={onKeyDown && onKeyDown}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:150]:
```suggestion
          onKeyDown={onKeyDown}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    return (
+      <StyledLabelWrapper className={className}>
+        <AminoInput
+          aria-label={label}
+          className={[
+            error && error.length ? 'has-error' : '',
+            label ? 'has-label' : '',
+            value ? 'has-content' : '',
+          ].join(' ')}
+          autoFocus={autoFocus && autoFocus}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:146]:
```suggestion
          autoFocus={autoFocus}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+  /** Input value. Required since all inputs must be fully controlled */
+  value: string | null;
+
+  /** Input on changed. Required since all inputs must be fully controlled */
+  onChange: React.ChangeEventHandler<HTMLInputElement>;
+
+  /** Placeholder text to be displayed in the input */
+  /** @deprecated Please use label instead. This placeholder will be replaced with float label. */
+  placeholder?: string;
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:90]: Let's chat about this one. I would expect `placeholder` to be the best name for the text that is displayed in the input.

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+            value ? 'has-content' : '',
+          ].join(' ')}
+          autoFocus={autoFocus && autoFocus}
+          disabled={disabled}
+          inputMode={inputMode}
+          onChange={onChange}
+          onKeyDown={onKeyDown && onKeyDown}
+          pattern={pattern && pattern}
+          placeholder={placeholder || label}
+          readOnly={readOnly || false}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:153]:
```suggestion
          readOnly={readOnly}
```

File: src/components/Input/InputType/FloatLabelInput.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+          ref={ref}
+          required={required || false}
```
Comment (goodpickle) [inline src/components/Input/InputType/FloatLabelInput.tsx:155]:
```suggestion
          required={required}
```

File: src/components/Input/InputType/NumberInput.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        autoFocus={autoFocus}
+        disabled={disabled}
+        error={error}
+        inputMode={inputMode}
+        label={label}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        pattern={pattern && pattern}
+        placeholder={placeholder}
+        readOnly={readOnly || false}
```
Comment (goodpickle) [inline src/components/Input/InputType/NumberInput.tsx:65]:
```suggestion
        readOnly={readOnly}
```

File: src/components/Input/InputType/PasswordInput.tsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+        autoFocus={autoFocus}
+        disabled={disabled}
+        error={error}
+        inputMode={inputMode}
+        label={label}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        pattern={pattern && pattern}
+        placeholder={placeholder}
+        readOnly={readOnly || false}
```
Comment (goodpickle) [inline src/components/Input/InputType/PasswordInput.tsx:73]:
```suggestion
        readOnly={readOnly}
```

File: src/components/Input/SearchInput.tsx [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+      </StyledLabel>
+      <AminoInput
+        autoFocus={autoFocus}
+        disabled={disabled}
+        inputMode={inputMode}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        pattern={pattern && pattern}
+        placeholder={placeholder || 'search...'}
+        readOnly={readOnly || false}
```
Comment (goodpickle) [inline src/components/Input/SearchInput.tsx:93]:
```suggestion
        readOnly={readOnly}
```

**Comment by Thienhuynh95** [comment]
@goodpickle I've just merged and fixed the conflict, can you please check again!

File: src/icons/IconBase copy.tsx [hunk 13]
```diff
@@ -0,0 +1,24 @@
+import React, { forwardRef, ReactNode } from 'react';
+
```
Comment (goodpickle) [inline src/icons/IconBase copy.tsx:2]: This component filename is `IconBase copy.tsx`

I'm guessing it is a file that needs to be removed or renamed. 🤓

**Review by goodpickle — CHANGES_REQUESTED** [review]
Most of my suggestions are just general cleanup for code that was copied and pasted.

**Review by goodpickle — APPROVED** [review]
Looks good. Let's chat about another solution instead of a `Link` button at some point.

File: src/components/Button/Link.tsx [hunk 14]
```diff
@@ -32,6 +32,7 @@ const AminoLink = styled.a<Pick<LinkProps, 'size' | 'disabled'>>`
 
   &.only-icon {
     width: 32px;
+    padding: 0;
```
[the remark this answers is not in the document]
Comment (Thienhuynh95) [reply src/components/Button/Link.tsx:35]: Gotcha, do you want me to move `Link` component from `dashboard-web` over?

[the remark this answers is not in the document]
Comment (goodpickle) [reply src/components/Button/Link.tsx:35]: No. I don't know what to do about it. I need to brainstorm on it. Probably come up with a different component name for this one.

**Pull request merged**